apriltag_ros/src/continuous_detector.cpp

```diff
@@ -86,6 +93,72 @@ bool ContinuousDetector::refreshParamsCallback(std_srvs::Empty::Request& req,
   return true;
 }
 
+void ContinuousDetector::publishEmptyDetection(const std_msgs::Header& header)
+{
+  AprilTagDetectionArray tag_detection_array;
+  tag_detection_array.header = header;
+  tag_detections_publisher_.publish(tag_detection_array);
+}
+
+bool ContinuousDetector::isTagInFOV(const tf::Transform& tag_pose) const
+{
+  if (!camera_model_.initialized())
+  {
+    ROS_WARN_STREAM_THROTTLE(1.0, "Camera not initialized, skipping FOV check");
+    return false;
+  }
+  if (tag_pose.getOrigin().getZ() <= 0)
+  {
+    // handles the case where Z is equal to zero (shouldn't happen) or less than zero (behind the camera)
+    return false;
+  }
+
+  // convert the 3d pose in the image frame to a 2d pixel frame
+  const auto image_pt = camera_model_.project3dToPixel(
+    cv::Point3f(
+      tag_pose.getOrigin().getX(), 
+      tag_pose.getOrigin().getY(), 
+      tag_pose.getOrigin().getZ()
+    )
+  );
+  ROS_DEBUG_STREAM_THROTTLE(1.0, "Frame " << camera_model_.cameraInfo().header.frame_id << " has target at pixel coordinate x: " << image_pt.x << ", y: " << image_pt.y);
+
+  // check if the image is within the target fov (from image width and height in pixels), return true if it is and false otherwise
+  return fov_pixel_buffer_width_ <= image_pt.x && image_pt.x < (camera_model_.cameraInfo().width - fov_pixel_buffer_width_) && fov_pixel_buffer_height_ <= image_pt.y && image_pt.y < (camera_model_.cameraInfo().height - fov_pixel_buffer_height_);
```

Comment:
It looks like this line and some below are two large.

It's probably a bad idea to add .clang-format to this repo, but you could at least split this one in two lines

Reply:
Ok - I'll check for long lines that I added. Agree that clang-format is a bad idea so that things don't change too much from upstream (especially if we want to rebase from upstream in the future).